h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -696,10 +698,65 @@ public void init(boolean expensive) {
         _parms._gradient_epsilon = _parms._lambda[0] == 0 ? 1e-6 : 1e-4;
         if(_parms._lambda_search) _parms._gradient_epsilon *= 1e-2;
       }
+
+      if (_parms.hasCheckpoint()) {
+        if (!_parms._family.equals(Family.gaussian))  // Gaussian it not iterative and therefore don't care
+          _checkPointFirstIter = true;  // mark the first iteration during iteration process of training
+        if (!_parms._solver.equals(Solver.IRLSM))
+          error("_checkpoint", "GLM checkpoint is supported only for IRLSM.  Please specify it " +
+                  "explicitly.  Do not use AUTO or default");
+        Value cv = DKV.get(_parms._checkpoint);
+        CheckpointUtils.getAndValidateCheckpointModel(this, GLMModel.GLMParameters.CHECKPOINT_NON_MODIFIABLE_FIELDS, cv);
```

Comment:
CHECKPOINT_NON_MODIFIABLE_FIELDS is empty, hence this does nothing, is that intentional?

Reply:
Honza:

I only implemented checkpointing for solver IRLSM.  Hence, I am leaving this here for future implementation that will use this.

Reply:
that would mean that at least _solver param should be amongst those params that should be the same as the ones in the checkpoint model. are you sure there are not other that should remain the same between the checkpoint and the new model?

Reply:
Honza: 

Great suggestion.  I will add it.  Thank you so much!

Wendy